### PR TITLE
feat(eks): aws-node-termination-handler for spot instances now pulls the image from public ECR

### LINKED ISSUE #12134: (eks): config aws-node-termination-handler image use public ECR
If you want to use spot worker node, `aws-eks` construct will help you install `aws-node-termination-handler`.
Now helm chart will pull image from docker hub by default.

see this repo: 
https://github.com/aws/aws-node-termination-handler/blob/7ec2ce5457fd7a8a3d4fe0fb3f95445cb1cde9fc/config/helm/aws-node-termination-handler/values.yaml#L5-L9

Since dockerhub has begun rate limiting pulls, it will  affect many CDK users with `aws-eks` construct.
https://github.com/aws/aws-cdk/blob/d5afb555b983c8c034f63dd58d1fa24b82b6e9fe/packages/%40aws-cdk/aws-eks/lib/cluster.ts#L1464-L1474

### Proposed Solution

see this PR: https://github.com/aws/aws-node-termination-handler/pull/325

`aws-node-termination-handler` can pull from https://gallery.ecr.aws/r6b0f9a1/aws-node-termination-handler now.
I will create the PR for this 
```ts
 if (!this._spotInterruptHandler) { 
   this._spotInterruptHandler = this.addHelmChart('spot-interrupt-handler', { 
     chart: 'aws-node-termination-handler', 
     version: '0.9.5', 
     repository: 'https://aws.github.io/eks-charts', 
     namespace: 'kube-system', 
     values: { 
       'nodeSelector.lifecycle': LifecycleLabel.SPOT,
       'image': {
            repository: 'public.ecr.aws/r6b0f9a1/aws-node-termination-handler',
          },
     }, 
   }); 
 } 
```

### Other

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by neilkuan**
close #12134 

update `aws-node-termination-handler` version to `0.13.2`
```ts
this._spotInterruptHandler = this.addHelmChart('spot-interrupt-handler', {
        chart: 'aws-node-termination-handler',
        version: '0.13.2',
        repository: 'https://aws.github.io/eks-charts',
        namespace: 'kube-system',
        values: {
          'nodeSelector.lifecycle': LifecycleLabel.SPOT,
        },
      });
```

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by guan840912** [comment]
`aws-node-termination-handler` just release version 0.13.2, config `repository: public.ecr.aws/r6b0f9a1/aws-node-termination-handler` by default .
https://github.com/aws/aws-node-termination-handler/blob/fe691872e75b737e19d0edd85db359091c996d63/config/helm/aws-node-termination-handler/values.yaml#L6
https://github.com/aws/aws-node-termination-handler/blob/v1.11.2/config/helm/aws-node-termination-handler/Chart.yaml

**Comment by guan840912** [comment]
waiting for https://github.com/aws/aws-node-termination-handler/actions/runs/429305845
release success...

**Comment by guan840912** [comment]
Hi @iliapolo , please review ;)
截圖 2020-12-24 上午9 55 51

**Comment by iliapolo** [comment]
@guan840912 So version `0.13.2` pulls from ECR by default?

Also notice there is a conflict.

**Comment by bwagner5** [comment]
Yes, aws-node-termination-handler helm chart v0.13.2, now, by default, pulls v1.11.2+ from the ECR public gallery.  

https://github.com/aws/eks-charts/commit/139cfc3488d8f3c827a8ccd3d8d2217279540b54

**Comment by guan840912** [comment]
Hi @iliapolo, fixed  the conflict :)

**Review by iliapolo — DISMISSED** [review]

File: packages/@aws-cdk/aws-eks/lib/cluster.ts [hunk 1]
```diff
@@ -1440,7 +1440,7 @@ export class Cluster extends ClusterBase {
     if (!this._spotInterruptHandler) {
       this._spotInterruptHandler = this.addHelmChart('spot-interrupt-handler', {
         chart: 'aws-node-termination-handler',
-        version: '0.9.5',
+        version: '0.13.2',
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-eks/lib/cluster.ts:1443]:
```suggestion
        version: '0.13.2', 
```

File: packages/@aws-cdk/aws-eks/lib/cluster.ts [hunk 2]
```diff
@@ -1440,7 +1440,7 @@ export class Cluster extends ClusterBase {
...
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-eks/lib/cluster.ts:1443]:
```suggestion
        version: '0.13.2',
```

**Review by iliapolo — APPROVED** [review]

**Pull request merged**